File: void_front/game/core/player/ships/enemies/AI.py
from __future__ import annotations

from game.core.assets.main import assets
from game.core.player.ships.enemies.enemy import EnemyShip
from game.core.player.ships.shapes import shapes
from game.core.weapons.plasma_canon import PlasmaCanon


class AI(EnemyShip):
    def __init__(self, x=10, y=20, angle=0):
        super().__init__(x=x, y=y, width=50, height=50, angle=angle, acceleration=1000, color="blue", vertices=shapes[4], name="Bomber Drone", maxWeaponHeat=10000, life=200, weapon=PlasmaCanon, img=getattr(getattr(assets, "images", None), "bombership", None), flameImg=getattr(getattr(assets, "images", None), "flame2", None))
        self.seekAcceleration = self.acceleration
        self.fleeAcceleration = self.acceleration * 0.9
        self.predicting = False
        self.state = "AI"

    def update(self, t, dt, thrust=0, turn=0):
        super().update(t, dt, thrust, turn)

File: core/player/ships/enemies/AI.js
import { assets } from "../../../assets/main.js";
import { nearByEnemy } from "../../../utils/distance.js";
import PlasmaCanon from "../../../weapons/plasmaCanon.js";
import { canvas } from "../../../world/canvas.js";
import { world } from "../../../world/world.js";
import { ship } from "../player.js";
import { shapes } from "../shapes.js";
import EnemyShip from "./enemy.js";
import { Tensor } from "onnxruntime-web";

export default class AI extends EnemyShip {
    constructor({ x = 10, y = 20, angle = 0 }) {
        super({ x, y, width: 50, height: 50, angle, acceleration: 10000, color: "blue", vertices: shapes[4], name: "Bomber Drone", maxWeaponHeat: 10000, life: 200, weapon: PlasmaCanon, img: assets?.images?.bombership, flameImg: assets?.images?.flame2, turnRate: 2 });
        this.seekAcceleration = this.acceleration;
        this.fleeAcceleration = this.acceleration * 0.9;
        this.predicting = false;

        console.log("AI ship created", this);
        this.state = "AI";
    }

    update(t, dt, thrust = 0, turn = 0) {
        super.update(t, dt, thrust, turn);
    }
}